CODEOWNERS

```diff
@@ -1,50 +1,71 @@
-/topics/ingest-performance-metrics.yaml                    @getsentry/owners-ingest @getsentry/owners-snuba
-/topics/ingest-replay-recordings.yaml                      @getsentry/owners-ingest @getsentry/owners-snuba @getsentry/replay
+/topics/ingest-metrics.yaml                                    @getsentry/owners-ingest @getsentry/owners-snuba
+/topics/ingest-performance-metrics.yaml                        @getsentry/owners-ingest @getsentry/owners-snuba
+/topics/ingest-replay-recordings.yaml                          @getsentry/owners-ingest @getsentry/owners-snuba @getsentry/replay
 
 # Topics consumed by Snuba
-/topics/events.yaml                                        @getsentry/owners-snuba
-/topics/transactions.yaml                                  @getsentry/owners-snuba
-/topics/outcomes.yaml                                      @getsentry/owners-snuba
-/topics/snuba-metrics.yaml                                 @getsentry/owners-snuba
-/topics/snuba-generic-metrics.yaml                         @getsentry/owners-snuba
-/topics/generic-events.yaml                                @getsentry/owners-snuba @getsentry/issues
-/topics/ingest-replay-events.yaml                          @getsentry/owners-snuba @getsentry/replay
-/topics/processed-profiles.yaml                            @getsentry/owners-snuba @getsentry/profiling
-/topics/profiles-call-tree.yaml                            @getsentry/owners-snuba @getsentry/profiling
-/topics/group-attributes.yaml                              @getsentry/owners-snuba @getsentry/issues
-/topics/snuba-spans.yaml                                   @getsentry/owners-snuba @getsentry/profiling
-/topics/snuba-metrics-summaries.yaml                       @getsentry/owners-snuba @getsentry/profiling
+/topics/events.yaml                                            @getsentry/owners-snuba
+/topics/transactions.yaml                                      @getsentry/owners-snuba
+/topics/outcomes.yaml                                          @getsentry/owners-snuba
+/topics/snuba-metrics.yaml                                     @getsentry/owners-snuba
+/topics/snuba-generic-metrics.yaml                             @getsentry/owners-snuba
+/topics/generic-events.yaml                                    @getsentry/owners-snuba @getsentry/issues
+/topics/ingest-replay-events.yaml                              @getsentry/owners-snuba @getsentry/replay
+/topics/processed-profiles.yaml                                @getsentry/owners-snuba @getsentry/profiling
+/topics/profiles-call-tree.yaml                                @getsentry/owners-snuba @getsentry/profiling
+/topics/group-attributes.yaml                                  @getsentry/owners-snuba @getsentry/issues
+/topics/snuba-spans.yaml                                       @getsentry/owners-snuba @getsentry/profiling
+/topics/snuba-metrics-summaries.yaml                           @getsentry/owners-snuba @getsentry/profiling
```

Comment:
Have you ever considered laying out the repo in order to allow product teams to have their own directory that contain topic, schema and examples so that code owners do not have to be updated for each new topic ?
I'd be curious to hear if that was considered and discarded.

Reply:
I haven't personally considered it. Though my feeling is that layout is probably good for codeowners and not a lot else. I think that layout comes with it's own challenges. You can see it in Snuba's dataset layout for example which uses something like what you mention. There I find it hard to find the right files when they are nested and it also means startup time for consumers (which only care about one storage) is pretty suboptimal as everything must always be loaded beforehand, and the code to load everything gets pretty complicated.